lua/wikis/commons/ThisDay.lua

```diff
@@ -0,0 +1,362 @@
+local ColumnName = Condition.ColumnName
+
+local OpponentLibraries = Lua.import('Module:OpponentLibraries')
+local Opponent = OpponentLibraries.Opponent
+local OpponentDisplay = OpponentLibraries.OpponentDisplay
+
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Link = Lua.import('Module:Widget/Basic/Link')
+local WidgetUtil = Lua.import('Module:Widget/Util')
+
+---@class ThisDayConfig
+---@field showPatches boolean?
+---@field showEmptyBirthdayList boolean?
+---@field showEmptyPatchList boolean?
+---@field showTrivia boolean?
+---@field tiers integer[]
+---@field tierTypes string[]
+
+local Config = Lua.import('Module:ThisDay/config', {loadData = true})
+
+local Query = {}
+
+--- Queries birthday data
+---@param month integer
+---@param day integer
+---@return player[]
+function Query.birthday(month, day)
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{
+			ConditionNode(ColumnName('birthdate_month'), Comparator.eq, month),
+			ConditionNode(ColumnName('birthdate_day'), Comparator.eq, day),
+			ConditionNode(ColumnName('deathdate'), Comparator.eq, DateExt.defaultDate),
+			ConditionNode(ColumnName('birthdate'), Comparator.neq, DateExt.defaultDate),
+		}
+
+	return mw.ext.LiquipediaDB.lpdb('player', {
+		limit = 5000,
+		conditions = conditions:toString(),
+		query = 'extradata, pagename, id, birthdate, nationality, links',
+		order = 'birthdate asc, id asc'
+	})
+end
+
+--- Queries patch data
+---@param month integer
+---@param day integer
+---@return datapoint[]
+function Query.patch(month, day)
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{
+			ConditionNode(ColumnName('date'), Comparator.neq, DateExt.defaultDate),
+			ConditionNode(ColumnName('date_month'), Comparator.eq, month),
+			ConditionNode(ColumnName('date_day'), Comparator.eq, day),
+			ConditionNode(ColumnName('type'), Comparator.eq, 'patch'),
+		}
+
+	return mw.ext.LiquipediaDB.lpdb('datapoint', {
+		limit = 5000,
+		conditions = conditions:toString(),
+		query = 'pagename, name, date',
+		order = 'date asc, name asc'
+	})
+end
+
+--- Queries tournament win data
+---@param month integer
+---@param day integer
+---@return placement[]
+function Query.tournament(month, day)
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{
+			ConditionNode(ColumnName('date'), Comparator.neq, DateExt.defaultDate),
+			ConditionNode(ColumnName('date_month'), Comparator.eq, month),
+			ConditionNode(ColumnName('date_day'), Comparator.eq, day),
+			ConditionNode(ColumnName('date'), Comparator.lt, os.date('%Y-%m-%d', os.time() - 86400)),
+			ConditionNode(ColumnName('placement'), Comparator.eq, 1),
+			ConditionNode(ColumnName('opponentname'), Comparator.neq, 'TBD'),
+			ConditionNode(ColumnName('opponentname'), Comparator.neq, 'Definitions'),
+			ConditionTree(BooleanOperator.any)
+				:add{ConditionNode(ColumnName('prizepoolindex'), Comparator.eq, '')}
+				:add{ConditionNode(ColumnName('prizepoolindex'), Comparator.eq, '1')}
```

Comment:
Only one of these should be needed as well

Reply:
yeah the `''` one was for legacy reasons during conversion